clients/src/test/java/org/apache/kafka/clients/NetworkClientTest.java

```diff
@@ -345,6 +372,18 @@ public void testCallDisconnect() throws Exception {
         assertTrue(client.canConnect(node, time.milliseconds()));
     }
 
+    private void awaitInFlightApiVersionRequest() throws Exception {
+        TestUtils.waitForCondition(new TestCondition() {
+            @Override
+            public boolean conditionMet() {
+                client.ready(node, time.milliseconds());
```

Comment:
Perhaps `ready()` can be outside `TestUtils.waitForCondition` since it is required only once?